Analysis/src/TypeInfer.cpp

```diff
@@ -4899,7 +4899,7 @@ TypeId TypeChecker::resolveType(const ScopePtr& scope, const AstType& annotation
             if (lit->parameters.size != 1 || !lit->parameters.data[0].type)
             {
                 reportError(TypeError{annotation.location, GenericError{"_luau_print requires one generic parameter"}});
-                return errorRecoveryType(anyType);
+                return currentModule->astResolvedTypes[&annotation] = errorRecoveryType(anyType);
```

Comment:
I decided to do this in all return places instead of storing the values in a `result` var and returning that (like in constraint graph builder), because there was a lot of short circuiting requires present. The alternative would have probably needed a restructuring of the control flow.

I'm not sure this is the best way to do it though, and there is a possibility of user error if you forget to add the result into the map.
If there is a better solution to this, or you want me to instead change to a `result` var, please let me know

Reply:
Something else you can do is turn this function into a helper, and forward the arguments. I don't think we have an example like this anywhere, but that's the cleanest solution. I'm not a fan of using assignments as expressions.

Reply:
I'm not entirely sure what you mean
Do you mean converting it into something like the overloaded check/checkExpr functions, and passing through the arguments to the correct overload?
Or creating like a pass-through helper like "recordResolvedType" so its now `return recordResolveType(annotation, ty)` which stores ty in the table and re-returns it.

Reply:
Sorry. The latter, but the other way around. So the end product looks like:

```cpp
TypeId TypeChecker::resolveType(const ScopePtr& scope, const AstType& annotation)
{
    TypeId& ty = currentModule->astResolvedTypes[&annotation];
    ty = resolveTypeWorker(scope, annotation);
    return ty;
}
```

And then rename the old `resolveType` function into `resolveTypeWorker` so that all call sites go through `resolveType`, and `resolveTypeWorker` simply focuses on building up the `TypeId`. Ideally, `resolveTypeWorker` is only ever called by `resolveType`.